include/aspect/utilities.h

```diff
@@ -1232,6 +1232,19 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double SPD_safety_factor);
 
+    /**
+     * Computes the current_edot_ii and current_stress, with the option for
+     * viscous and viscoelastic contribution.
+     */
+    template <int dim>
+    double compute_current_edot_ii (const unsigned int j,  // the volume fraction
```

Comment:
I'm currently reading "Code Complete" by Steve McConnell with all of my students, which has a lot of good advice on function and argument names. So I'll comment on this function.

For `j`, if you need to document the name of an argument, the name is probably poorly chosen :-) If the argument really denoted the volume fraction, then `volume_fraction` would be a better argument name. But that can't be true, because it is an integer (and the volume fraction should be a `double`), so it must be something different -- I don't know what it is, but it would be useful to choose a more descriptive name than `j` :-)

Reply:
yes, I see my comment about j was misleading. So j comes from this for loop: 
for (unsigned int j=0; j < volume_fractions.size(); ++j)
which is way I made the little comment for myself that it is not from another loop through lets say n_evaluation_points